### PR TITLE
[Issue #47744] Add failing unit test depicting off-by-one error

### PR DESCRIPTION
**Pull request by dmg-aimmerman**
Please review original issue here: https://github.com/laravel/framework/issues/47744

Issue as reported by QA team:
image

### CODE REVIEW

File: tests/Validation/ValidationValidatorTest.php [hunk 1]
```diff
@@ -728,6 +728,33 @@ public function testPositionValuesAreReplaced()
...
 
+    public function testArrayValidationMessagesAreAccurate()
+    {
+        // Note that the *second* email is invalid in the user input sample
+        $userInput = ['test@example.com', 'invalid_email', 'john@test.com'];
+
+        // Since the issue occurs between module boundaries (validation &
+        //  locale), we'll need to load the actual locale file.
+        $trans = $this->getIlluminateArrayTranslator();
+        $locale = require_once('src/Illuminate/Translation/lang/en/validation.php');
```
Comment (GrahamCampbell) [inline tests/Validation/ValidationValidatorTest.php:739]: require_once is the wrong function here, as running the test twice (with --times 2) will fail

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Validation/ValidationValidatorTest.php (hunk 1 above)
Comment (GrahamCampbell) [reply tests/Validation/ValidationValidatorTest.php:739]: requiring the file at all is still pretty obscure tbh

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: tests/Validation/ValidationValidatorTest.php (hunk 1 above)
Comment (dmg-aimmerman) [reply tests/Validation/ValidationValidatorTest.php:739]: @GrahamCampbell Absolutely agreed, thank you for the great comments. Looking at the existing tests for validation, it doesn't appear that any actually validate the actual error messages. Do you have any suggestions on a better way to load the locale in the context of unit tests? Or should I just change `require_once` to just `require`?

Side note, this PR is for demonstration purposeses (for now at least) as requested on the original issue report.

**Comment by taylorotwell** [comment]
Seems there is some discussion as to whether this is a bug - will follow up on issue thread.

**Pull request closed without merging**